app/src/main/java/io/github/muntashirakon/setedit/Utils/EditorUtils.java

```diff
@@ -24,22 +26,38 @@
 
 import java.util.List;
 
+import io.github.muntashirakon.setedit.EditorActivity;
+import io.github.muntashirakon.setedit.R;
+
 public class EditorUtils {
+
+
     /**
-     * Check whether the permission has been granted
+     * Check whether the settings write permission has been granted
      *
      * @return {@code true} if granted, {@code null} if is being granted and {@code false} otherwise
      */
     @Nullable
-    public static Boolean checkPermission(@NonNull Context context, @NonNull String tableType) {
+    public static Boolean checkSettingsWritePermission(@NonNull Context context, @NonNull String tableType) {
         String permission = "system".equals(tableType) ? Manifest.permission.WRITE_SETTINGS : Manifest.permission.WRITE_SECURE_SETTINGS;
         if ("system".equals(tableType) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.System.canWrite(context)) {
-            try {
-                Intent intent = new Intent(Settings.ACTION_MANAGE_WRITE_SETTINGS)
-                        .setData(Uri.parse("package:" + BuildConfig.APPLICATION_ID));
-                context.startActivity(intent);
-            } catch (Exception ignore) {
-            }
+            new AlertDialog.Builder(context)
+                    .setTitle("Permission needed")
+                    .setMessage("You have to enable SetEditPlus for Modify system settings")
+                    .setPositiveButton("ok", (dialog, which) -> {
+                        try {
+                            Intent intent = new Intent(Settings.ACTION_MANAGE_WRITE_SETTINGS)
+                                    .setData(Uri.parse("package:" + context.getPackageName()));
```

Comment:
`BuildConfig.APPLICATION_ID` is now preferred over `Context#getPackageName()`.

Reply:
Fixed